Source/WebCore/rendering/GridTrackSizingAlgorithm.cpp

```diff
@@ -1388,37 +1388,56 @@ static LayoutUnit computeSubgridMarginBorderPadding(const RenderGrid* outermost,
+    auto gutterTotal = 0_lu;
 
-        for (auto& currentAncestorSubgrid : ancestorSubgridsOfGridItem(*gridItem, flowAwareDirection)) {
-            std::optional<LayoutUnit> availableSpace;
-            if (!GridLayoutFunctions::hasRelativeOrIntrinsicSizeForChild(currentAncestorSubgrid, flowAwareDirection))
-                availableSpace = currentAncestorSubgrid.availableSpaceForGutters(flowAwareDirection);
+    for (auto& currentAncestorSubgrid : ancestorSubgridsOfGridItem(gridItem, direction)) {
+        std::optional<LayoutUnit> availableSpace;
+        if (!GridLayoutFunctions::hasRelativeOrIntrinsicSizeForChild(currentAncestorSubgrid, direction))
+            availableSpace = currentAncestorSubgrid.availableSpaceForGutters(direction);
 
-            auto gridItemSpanInAncestor = currentAncestorSubgrid.gridSpanForChild(*gridItem, flowAwareDirection);
-            auto numTracksForCurrentAncestor = currentAncestorSubgrid.numTracks(flowAwareDirection);
+        auto gridItemSpanInAncestor = currentAncestorSubgrid.gridSpanForChild(gridItem, direction);
+        auto numTracksForCurrentAncestor = currentAncestorSubgrid.numTracks(direction);
 
-            const auto* currentAncestorSubgridParent = dynamicDowncast<RenderGrid>(currentAncestorSubgrid.parent());
-            ASSERT(currentAncestorSubgridParent);
-            if (!currentAncestorSubgridParent)
-                return std::nullopt;
+        const auto* currentAncestorSubgridParent = dynamicDowncast<RenderGrid>(currentAncestorSubgrid.parent());
+        ASSERT(currentAncestorSubgridParent);
+        if (!currentAncestorSubgridParent)
+            return std::nullopt;
 
-            if (gridItemSpanInAncestor.startLine())
-                gutterTotal += (currentAncestorSubgrid.gridGap(flowAwareDirection) - currentAncestorSubgridParent->gridGap(flowAwareDirection)) / 2;
-            if (span.endLine() != numTracksForCurrentAncestor)
-                gutterTotal += (currentAncestorSubgrid.gridGap(flowAwareDirection) - currentAncestorSubgridParent->gridGap(flowAwareDirection)) / 2;
-            flowAwareDirection = GridLayoutFunctions::flowAwareDirectionForParent(currentAncestorSubgrid, *currentAncestorSubgridParent, flowAwareDirection);
-        }
-        return gutterTotal;
-    };
+        if (gridItemSpanInAncestor.startLine())
+            gutterTotal += (currentAncestorSubgrid.gridGap(direction) - currentAncestorSubgridParent->gridGap(direction)) / 2;
+        if (itemSpan.endLine() != numTracksForCurrentAncestor)
+            gutterTotal += (currentAncestorSubgrid.gridGap(direction) - currentAncestorSubgridParent->gridGap(direction)) / 2;
+        direction = GridLayoutFunctions::flowAwareDirectionForParent(currentAncestorSubgrid, *currentAncestorSubgridParent, direction);
+    }
+    return gutterTotal;
+}
+
+bool GridTrackSizingAlgorithm::shouldExcludeGridItemForMasonryTrackSizing(const RenderBox& gridItem, unsigned trackIndex, GridSpan itemSpan) const
```

Comment:
Maybe we should consider inlining these functions?

Reply:
Not sure! It would depend on how hot these functions are and right now I don't know

Reply:
We should let the compiler decides about inlining (except for accessors by the guidelines)